ThirdPartyAdapters/inmobi/inmobi/src/main/java/com/google/ads/mediation/inmobi/InMobiUnifiedNativeAdMapper.java

```diff
@@ -79,154 +85,143 @@ public InMobiUnifiedNativeAdMapper(@NonNull InMobiNativeWrapper inMobiNativeWrap
+              Log.w(TAG, error.toString());
+              mediationAdLoadCallback.onFailure(error);
+            }
+          }
 
-        placeHolderView.addView(primaryView);
-        int viewHeight = primaryView.getLayoutParams().height;
-        if (viewHeight > 0) {
-          setMediaContentAspectRatio((float) primaryView.getLayoutParams().width / viewHeight);
-        }
-      }
-    });
-
-    setMediaView(placeHolderView);
-    boolean hasVideo = (inMobiNativeWrapper.isVideo() == null) ? false : inMobiNativeWrapper.isVideo();
-    setHasVideoContent(hasVideo);
-
-    // Download drawables.
-    if (!this.isOnlyURL) {
-      new ImageDownloaderAsyncTask(new ImageDownloaderAsyncTask.DrawableDownloadListener() {
-        @Override
-        public void onDownloadSuccess(HashMap<String, Drawable> drawableMap) {
-          Drawable iconDrawable = drawableMap.get(ImageDownloaderAsyncTask.KEY_ICON);
-          setIcon(new InMobiNativeMappedImage(iconDrawable, iconUri, iconScale));
-
-          List<NativeAd.Image> imagesList = new ArrayList<>();
-          imagesList.add(
-              new InMobiNativeMappedImage(new ColorDrawable(Color.TRANSPARENT), null, 1.0));
-          setImages(imagesList);
-
-          if (null != iconDrawable && mediationAdLoadCallback != null) {
-            inMobiNativeAd.mediationNativeAdCallback = mediationAdLoadCallback.onSuccess(
-                InMobiUnifiedNativeAdMapper.this);
-          } else {
+          @Override
+          public void onDownloadFailure() {
             AdError error = InMobiConstants.createAdapterError(ERROR_NATIVE_ASSET_DOWNLOAD_FAILED,
-                "InMobi SDK failed to download native ad image assets.");
+                    "InMobi SDK failed to download native ad image assets.");
             Log.w(TAG, error.toString());
             mediationAdLoadCallback.onFailure(error);
           }
+        }).execute(map);
+      } else {
+        if (mediationAdLoadCallback != null) {
+          inMobiNativeAd.mediationNativeAdCallback = mediationAdLoadCallback.onSuccess(
+                  InMobiUnifiedNativeAdMapper.this);
         }
-
-        @Override
-        public void onDownloadFailure() {
-          AdError error = InMobiConstants.createAdapterError(ERROR_NATIVE_ASSET_DOWNLOAD_FAILED,
-              "InMobi SDK failed to download native ad image assets.");
-          Log.w(TAG, error.toString());
-          mediationAdLoadCallback.onFailure(error);
-        }
-      }).execute(map);
-    } else {
-      if (mediationAdLoadCallback != null) {
-        inMobiNativeAd.mediationNativeAdCallback = mediationAdLoadCallback.onSuccess(
-            InMobiUnifiedNativeAdMapper.this);
       }
+
     }
   }
 
-  @Override
-  public void handleClick(View view) {
-    // Handle click.
-    inMobiNativeWrapper.reportAdClickAndOpenLandingPage();
-  }
 
   @Override
   public void untrackView(View view) {
-    inMobiNativeWrapper.destroy();
+    inMobiNativeWrapper.unTrackViews();
```

Comment:
Does calling untrackViews will internally call destroy, because previously it used to destroy